src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  constructor() {
+    super();
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+
+    Animal.alive.push(this);
+  }
+
+  bite(herb) {
+    if (herb.hidden === false) {
+      herb.health -= 50;
+    }
```

Comment:
It's necessary to check that the `herb` argument is an instance of `Herbivore` before attempting to bite. If `herb` is not a `Herbivore`, the method should not proceed with the biting action. This is to ensure that a `Carnivore` can only bite a `Herbivore` and not another `Carnivore` or any other kind of object.